arangod/RocksDBEngine/ReplicatedRocksDBTransactionState.cpp

```diff
@@ -183,20 +183,42 @@ futures::Future<Result> ReplicatedRocksDBTransactionState::doCommit() {
       if (partialResults.empty() || partialResults.front().ok()) {
         return {};
       }
-      LOG_TOPIC("6d1ce", ERR, Logger::REPLICATED_STATE)
-          << "The leader has resigned, so nothing has been replicated. "
-             "However, if the transaction spans multiple collections, with "
-             "different leaders, it might have been partially applied.";
+
+      if (std::all_of(
+              partialResults.begin(), partialResults.end(), [](auto const& r) {
+                return r.is(
+                    TRI_ERROR_REPLICATION_REPLICATED_LOG_LEADER_RESIGNED);
+              })) {
+        return Result{
+            TRI_ERROR_REPLICATION_REPLICATED_LOG_LEADER_RESIGNED,
+            fmt::format(
+                "All the replicated log leaders involved in transaction {} "
+                "have resigned before the commit operation could be "
+                "replicated. The transaction has neither been committed "
+                "locally nor replicated, and it is going to be aborted.",
+                tid.asCoordinatorTransactionId().id())};
+      }
+
+      auto warningMsg = fmt::format(
+          "Some replicated log leaders have resigned before replicating "
+          "the commit operation of transaction {}. The transaction may "
+          "have been successfully applied only on some of the shards.",
+          tid.asCoordinatorTransactionId().id());
+      LOG_TOPIC("6d1ce", ERR, Logger::REPLICATED_STATE) << warningMsg;
+      // There's no need to crash in production, since there's not much we can
+      // do.
       TRI_ASSERT(false) << partialResults;
-      // There's no need to crash in production.
-      return partialResults.front();
+      return {TRI_ERROR_TRANSACTION_INTERNAL, warningMsg};
     }
 
     LOG_TOPIC("8ebc0", FATAL, Logger::REPLICATED_STATE)
         << "Failed to commit replicated transaction locally (partial "
            "commits detected): "
         << partialResults;
     TRI_ASSERT(false) << partialResults;
+    // The leader is our of sync.
+    // It makes sense to crash here, because the  in the hopes that this server
```

Comment:
There is some reason missing in the comment

Reply:
Th reason is explained above, this is a leftover from when I re-formulated. Fixed in https://github.com/arangodb/arangodb/pull/20294/commits/8420831659e8125613b09069c0dd00a3355d7e00